android/java/org/chromium/chrome/browser/crypto_wallet/fragments/TransactionDetailsSheetFragment.java

```diff
@@ -0,0 +1,222 @@
+import org.chromium.chrome.browser.crypto_wallet.util.Utils;
+import org.chromium.chrome.browser.crypto_wallet.util.WalletConstants;
+import org.chromium.chrome.browser.crypto_wallet.util.WalletUtils;
+import org.chromium.url.GURL;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class TransactionDetailsSheetFragment extends BottomSheetDialogFragment {
+    public static final String TAG = TransactionDetailsSheetFragment.class.getName();
+
+    private ExecutorService mExecutor;
+    private Handler mHandler;
+    private WalletModel mWalletModel;
+    private WalletListItemModel mWalletListItemModel;
+    private TransactionInfo mTxInfo;
+    private ParsedTransaction mParsedTx;
+
+    public static TransactionDetailsSheetFragment newInstance(
+            WalletListItemModel walletListItemModel) {
+        return new TransactionDetailsSheetFragment(walletListItemModel);
+    }
+
+    private TransactionDetailsSheetFragment(WalletListItemModel walletListItemModel) {
+        mWalletListItemModel = walletListItemModel;
+        mTxInfo = mWalletListItemModel.getTransactionInfo();
+        mParsedTx = mWalletListItemModel.getParsedTx();
+        mExecutor = Executors.newSingleThreadExecutor();
+        mHandler = new Handler(Looper.getMainLooper());
+    }
+
+    @Override
+    public void show(FragmentManager manager, String tag) {
+        try {
+            TransactionDetailsSheetFragment fragment =
+                    (TransactionDetailsSheetFragment) manager.findFragmentByTag(
+                            TransactionDetailsSheetFragment.TAG);
+            FragmentTransaction transaction = manager.beginTransaction();
+            if (fragment != null) {
+                transaction.remove(fragment);
+            }
+            transaction.add(this, tag);
+            transaction.commitAllowingStateLoss();
+        } catch (IllegalStateException e) {
+            Log.e(TAG, "show", e);
+        }
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        try {
+            BraveActivity activity = BraveActivity.getBraveActivity();
+            mWalletModel = activity.getWalletModel();
+        } catch (BraveActivity.BraveActivityNotFoundException e) {
+            Log.e(TAG, "onCreate ", e);
+        }
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
+            @Nullable Bundle savedInstanceState) {
+        return LayoutInflater.from(getContext())
+                .inflate(R.layout.tx_details_bottom_sheet, container, false);
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        if (JavaUtils.anyNull(mWalletModel)) return;
```

Comment:
Why `anyNull`? I think `WalletModel` is not iterable?

Reply:
It is a null check for `mWalletModel`, nothing related to iteration here.

Reply:
Yes I understand that, I just feel it is unnecessary to call the function since there is only one object. Of course it still works this way, just a minor thing.